test/deps/binaryen/binaryen_test.cpp

```diff
@@ -84,3 +86,48 @@ TEST(BinaryenTest, Example1) {
   // dispose module
   delete wasm;
 }
+
+/**
+ * @given WebAssembly S-expression code exporting a function
+ * (sumtwo) taking two arguments of type i32 returning their sum of type i32
+ * @when this code is interpreted using Binaryen
+ * @then sumtwo implementation on wasm is invoked from C++ with given arguments
+ */
+TEST(BinaryenTest, InvokeWebAssemblyFunctionFromCpp) {
+  // wast code with imported function's call
+  std::string sexpr =
+      "(module "
+      "  (type $t0 (func (param i32 i32) (result i32)))"
+      "  (export \"sumtwo\" (func $sumtwo))"
+      "  (func $sumtwo (; 1 ;) (type $t0) (param $p0 i32) (param $p1 i32) "
+      "(result i32)"
+      "    (i32.add"
+      "      (local.get $p0)"
+      "      (local.get $p1)))"
+      ")";
+  // parse wast
+  SExpressionParser parser(const_cast<char *>(sexpr.c_str()));
```

Comment:
since we are using c++17:

`const_cast<char *>(sexpr.c_str())` -> `sexpr.data()`

Or even declare `char sexpr[]` and pass just the variable

Reply:
Makes sense.